snapcraft/internal/parser.py

```diff
@@ -204,45 +205,44 @@ class Options():
 
 
 def run(args):
-
-    path = args.get("--output")
+    path = args.get('--output')
     if path is None:
         path = PARTS_FILE
 
     index = args.get('--index')
     if index:
-        if "://" not in index:
-            index = "%s%s" % ("file://", os.path.join(os.getcwd(), index))
+        if '://' not in index:
+            index = '{}{}'.format(
+                'file://', os.path.join(os.getcwd(), index))
         output = urllib.request.urlopen(index).read()
     else:
         # XXX: fetch the index from the wiki
-        output = "{}"
+        output = '{}'
 
     master_parts_list = _process_index(output)
-
-
     _write_parts_list(path, master_parts_list)
 
-
     if args['--debug']:
         print(yaml.dump(master_parts_list, default_flow_style=False))
 
     return args
 
+
 def is_valid_parts_list(parts_list, parts):
     for partname in parts:
         if partname not in parts_list.keys():
-            logging.error(
-                "Part '%s' is missing from the parts entry" % partname)
+            logging.error('Part {!r} is missing from the parts entry'.format(
+                partname))
             return False
 
     return True
 
+
 def _write_parts_list(path, master_parts_list):
-    logging.debug("Writing parts list to %s", path)
-    with open(path, "w") as fp:
+    logging.debug('Writing parts list to {!r}'.format(path))
+    with open(path, 'w') as fp:
         fp.write(yaml.dump(master_parts_list,
-                            default_flow_style=False))
+                 default_flow_style=False))
```

Comment:
ah, either multiple of 4 or right nder the paren. static tests failed here and work now though